vm/src/stdlib/os.rs

```diff
@@ -1127,6 +1127,87 @@ pub(super) mod _os {
         OutputMode::String.process_path(curdir_inner(vm)?, vm)
     }
 
+    #[cfg(unix)]
+    #[pyfunction]
+    fn register_at_fork(kwargs: crate::function::KwArgs, vm: &VirtualMachine) -> PyResult<()> {
+        let mut match_found = false; // better way to handle this?
+        for (key, value) in kwargs.into_iter() {
+            if key == "before" {
+                match_found = true;
+                vm.state.before_forkers.lock().push(value.clone());
+            }
+            if key == "after_in_parent" {
+                match_found = true;
+                vm.state.after_forkers_parent.lock().push(value.clone());
+            }
+            if key == "after_in_child" {
+                match_found = true;
+                vm.state.after_forkers_child.lock().push(value.clone());
+            }
+        }
+
+        if !match_found {
+            return Err(vm.new_value_error("At least one argument is required.".to_owned()));
+        }
+
+        Ok(())
+    }
+
+    #[cfg(unix)]
+    fn run_at_forkers(funcs: Vec<PyObjectRef>, vm: &VirtualMachine) {
+        if !funcs.is_empty() {
+            for func in funcs.into_iter().rev() {
+                if let Err(e) = func.call((), vm) {
+                    let exit = e.fast_isinstance(vm.ctx.exceptions.system_exit);
+                    vm.run_unraisable(e, Some("Exception ignored in".to_owned()), func);
+                    if exit {
+                        // Do nothing!
+                    }
+                }
+            }
+        }
+    }
+
+    #[cfg(unix)]
+    fn py_os_before_fork(vm: &VirtualMachine) -> PyResult<()> {
+        let before_forkers: Vec<PyObjectRef> = std::mem::take(&mut *vm.state.before_forkers.lock());
+        run_at_forkers(before_forkers, vm);
+
+        Ok(())
+    }
+
+    #[cfg(unix)]
+    fn py_os_after_fork_child(vm: &VirtualMachine) -> PyResult<()> {
+        let after_forkers_child: Vec<PyObjectRef> =
+            std::mem::take(&mut *vm.state.after_forkers_child.lock());
+        run_at_forkers(after_forkers_child, vm);
+        Ok(())
+    }
+
+    #[cfg(unix)]
+    fn py_os_after_fork_parent(vm: &VirtualMachine) -> PyResult<()> {
+        let after_forkers_parent: Vec<PyObjectRef> =
+            std::mem::take(&mut *vm.state.after_forkers_parent.lock());
+        run_at_forkers(after_forkers_parent, vm);
+        Ok(())
+    }
+
+    #[cfg(unix)]
+    #[pyfunction]
+    fn fork(vm: &VirtualMachine) -> PyResult {
```

Comment:
Is the `#[cfg(unix)]` insufficient for the `test.support.requires_fork()` or the `requires_os_func('fork')` decorators to work?

Reply:
#[cfg(unix)] should be sufficient for test.support.requires_fork(). Checking this